one_fm/hiring/utils.py

```diff
@@ -314,24 +314,25 @@ def generate_employee_id(doc):
 
 
 def create_leave_policy_assignment(doc):
-	'''
+    """
 		Method to create Leave Policy Assignment for an Employee, if employee have a Leave Policy
 		Create Leave Policy based on Joining Date
 		args:
 			doc: Employee Object
-	'''
-	if doc.leave_policy:
-		assignment = frappe.new_doc("Leave Policy Assignment")
-		assignment.employee = doc.name
-		assignment.assignment_based_on = 'Joining Date'
-		assignment.leave_policy = doc.leave_policy
-		assignment.effective_from = doc.date_of_joining
-		# effective_to is an year of addition to effective_from
-		assignment.effective_to = getdate(add_days(add_years(doc.date_of_joining, 1), -1))
-		assignment.carry_forward = True
-		assignment.leaves_allocated = False
-		assignment.save()
-		assignment.submit()
+    """
+    if doc.leave_policy:
+        assignment = frappe.new_doc("Leave Policy Assignment")
+        assignment.employee = doc.name
+        assignment.assignment_based_on = 'Joining Date'
+        assignment.leave_policy = doc.leave_policy
+        assignment.effective_from = doc.date_of_joining
+        # effective_to is an year of addition to effective_from
+        assignment.effective_to = getdate(add_days(add_years(doc.date_of_joining, 1), -1))
+        assignment.carry_forward = True
+        assignment.leaves_allocated = False
+        assignment.save()
+        assignment.submit()
+        grant_leave_alloc_for_employee(assignment)
```

Comment:
grant_leave_alloc_for_employee will be called on submit of leave policy assignment and it is customised on LeavePolicyAssignmentOverride

Reply:
@pjamsheer that has not been allocating since last year.

Reply:
Tested this and i worked, you can test at your end as well

Reply:
It is confirmed from the HR that, Leave Policy Assignment and Leave Allocation are creating on creation of Employee.
The method `create_leave_policy_assignment` is called from the after insert of employee record.